File: Frontend/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  description?: string | null;
  status: string;
  priority: string;
  startDate?: string | null;
  dueDate?: string | null;
  createdAt: string;
  updatedAt?: string | null;
  ownerId: number;
  ownerName?: string | null;
  requestedById?: number | null;
  requestedByName?: string | null;
  customerId?: number | null;
  customerName?: string | null;
  teamId?: number | null;
  teamName?: string | null;
}

export interface CreateProjectRequest {
  name: string;
  description?: string;
  status: string;
  priority: string;
  startDate?: string | null;
  dueDate?: string | null;
  customerId?: number | null;
  teamId?: number | null;
  ownerId: number;
}

export interface UpdateProjectRequest {
  name: string;
  description?: string;
  status: string;
  priority: string;
  startDate?: string | null;
  dueDate?: string | null;
  customerId?: number | null;
  teamId?: number | null;
  ownerId: number;
}

File: Backend/CRM.API/CRM.API/Models/Project.cs
namespace CRM.API.Models
{
    public class Project
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public string Status { get; set; } = "Planning";

        public string Priority { get; set; } = "Medium";

        public DateTime? StartDate { get; set; }

        public DateTime? DueDate { get; set; }

        public int OwnerId { get; set; }

        public int? RequestedById { get; set; }

        public int? CustomerId { get; set; }

        public int? TeamId { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // ==========================================
        // Relationships
        // ==========================================

        public User Owner { get; set; } = null!;

        public User? RequestedBy { get; set; }

        public Customer? Customer { get; set; }

        public Team? Team { get; set; }

        public ICollection<ProjectMember> Members { get; set; }
            = new List<ProjectMember>();

        public ICollection<TaskItem> Tasks { get; set; }
            = new List<TaskItem>();
    }
}